### DISCUSSION TITLE
Use Git Tag instead of commit SHA when setting target revisions in argocd-update

### DISCUSSION THREAD

**Discussion by gnadaban** [akuity/kargo#3001] [General]
When using a Git Warehouse with created tags as freight, `argocd-update` should use the tag for desired revision.
Since Kargo 1.0 this is not the case.

This has a few negative side-effects:
1. Tooling that expects target revisions to be pinned to tags no longer work
2. Operators looking at ArgoCD applications through UIs and APIs would have to trace back the commit SHA to the repository, which in some cases might be not authorized
3. Notifications and alerts about applications become cryptic

It would be great if either as an option, or as a default behaviour the `argocd-update` promotion would be capable of using the tag as desired revision as before.

**Comment by krancour** [comment]
You'll find this already sorted out in main and the upcoming 1.1 release. The addition of support for an expression language, together with a new option on this step give you more flexibility in how target revision is set.
